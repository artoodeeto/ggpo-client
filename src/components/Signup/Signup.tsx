import React from 'react';

const Signup = () => {
  return (
    <div>
      <div>
        <form>
          <label htmlFor="email">
            Email:
            <input placeholder="youremail@address.com" type="text" name="email" id="" required />
          </label>

          <label htmlFor="username">
            Username:
            <input placeholder="user name" type="text" name="username" id="" required />
          </label>

          <label htmlFor="password">
            Password:
            <input placeholder="password" type="text" name="email" id="" required />
          </label>

          <input type="submit" value="Login" />
        </form>
      </div>
    </div>
  );
};

export default Signup;
