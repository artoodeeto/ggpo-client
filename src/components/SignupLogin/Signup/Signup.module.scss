@import '../../../assets/media-queries.scss';

.Signup__Form {
  display: flex;
  flex-direction: column;
  align-items: center;

  > input {
    font-size: 1em;
    margin: 0.2em;
  }
}
