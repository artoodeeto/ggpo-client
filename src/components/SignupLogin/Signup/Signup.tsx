import React, { useState, useEffect, FormEvent, FC } from 'react';
import * as userSelectors from 'store/user/Selectors';
import * as sessionSelectors from 'store/session/Selectors';
import * as sessionActions from 'store/session/Actions';
import { connect } from 'react-redux';
import { IState } from 'interfaces/stateInterface';
import { ILoginSignUpFormParams } from 'interfaces/session';

const onSubmitSignup = (
  event: FormEvent,
  email: string,
  username: string,
  password: string,
  onSignup: Function
): void => {
  event.preventDefault();

  const formSignup: ILoginSignUpFormParams = {
    username,
    email,
    password
  };

  onSignup(formSignup);
};

export const Signup: FC = (props: any) => {
  const [email, setEmail] = useState('');
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');

  useEffect(() => {}, [username, email, password]);

  return (
    <div>
      <div>
        <form onSubmit={(e) => onSubmitSignup(e, email, username, password, props.onSignup)}>
          <label htmlFor="email">
            Email:
            <input
              onChange={(e) => setEmail(e.target.value)}
              value={email}
              placeholder="youremail@address.com"
              type="text"
              name="email"
              id=""
              required
            />
          </label>

          <label htmlFor="username">
            Username:
            <input
              onChange={(e) => setUsername(e.target.value)}
              value={username}
              placeholder="user name"
              type="text"
              name="username"
              id=""
              required
            />
          </label>

          <label htmlFor="password">
            Password:
            <input
              onChange={(e) => setPassword(e.target.value)}
              value={password}
              placeholder="password"
              type="text"
              name="password"
              id=""
              required
            />
          </label>

          <input type="submit" value="Sigup" />
        </form>
      </div>
    </div>
  );
};

const mapStateToProps = (state: IState) => {
  return {
    sign: 'up',
    isAuthenticated: sessionSelectors.isUserAuthorized(state),
    userInfo: userSelectors.userInfo(state)
  };
};

const mapDispatchToProps = (dispatch: Function) => {
  return {
    onSignup: (userSignup: ILoginSignUpFormParams) => dispatch(sessionActions.signMeUp(userSignup))
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Signup);
