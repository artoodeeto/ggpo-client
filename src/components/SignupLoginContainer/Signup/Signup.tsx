import React, { useState, useEffect, FormEvent, FC } from 'react';
import { LoginSignUp } from '../../../interfaces/signupLogin';
import * as userSelectors from '../../../models/User/selectors';
import * as sessionActions from '../Store/Actions';
import { connect } from 'react-redux';
import { State } from '../../../interfaces/stateInterface';

const onSubmitSignup = (
  event: FormEvent,
  email: string,
  username: string,
  password: string,
  onSignup: Function
): void => {
  event.preventDefault();

  const formSignup: LoginSignUp = {
    username,
    email,
    password
  };

  onSignup(formSignup);
};

const Signup: FC = (props: any) => {
  const [email, setEmail] = useState('');
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');

  useEffect(() => {
    // console.log({ email, username, password });
  });

  return (
    <div>
      <div>
        <h1>{props.isAuthenticated ? props.foo : 'nope'}</h1>
        <form onSubmit={(e) => onSubmitSignup(e, email, username, password, props.onSignup)}>
          <label htmlFor="email">
            Email:
            <input
              onChange={(e) => setEmail(e.target.value)}
              value={email}
              placeholder="youremail@address.com"
              type="text"
              name="email"
              id=""
              required
            />
          </label>

          <label htmlFor="username">
            Username:
            <input
              onChange={(e) => setUsername(e.target.value)}
              value={username}
              placeholder="user name"
              type="text"
              name="username"
              id=""
              required
            />
          </label>

          <label htmlFor="password">
            Password:
            <input
              onChange={(e) => setPassword(e.target.value)}
              value={password}
              placeholder="password"
              type="text"
              name="password"
              id=""
              required
            />
          </label>

          <input type="submit" value="Sigup" />
        </form>
      </div>
    </div>
  );
};

const mapStateToProps = (state: State) => {
  return {
    sign: 'up',
    isAuthenticated: userSelectors.isUserAuthorized(state),
    userInfo: userSelectors.userInfo(state)
  };
};

const mapDispatchToProps = (dispatch: Function) => {
  return {
    onSignup: (userSignup: LoginSignUp) => dispatch(sessionActions.signMeUp(userSignup)),
    onLogout: () => dispatch(sessionActions.logMeOut())
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Signup);
