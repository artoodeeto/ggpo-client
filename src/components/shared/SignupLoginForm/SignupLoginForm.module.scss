@import '../../../assets/media-queries.scss';
@import '../../../assets/color-theme-dark.scss';

.SignUpLoginForm {
  display: flex;
  flex-direction: column;
  align-items: center;

  > input {
    box-sizing: border-box;
    width: 100%;
    padding: 12px;
    font-size: 14px;
    margin: 0.2em;
    border-radius: 4px;
    border: 0px;
  }

  > input[type='submit'] {
    margin-top: 5px;
    background-color: $blue-cheese;
    font-weight: bolder;
    font-size: 0.8em;
    color: $ashy-foot;
  }

  > input[type='email']:not(:placeholder-shown):invalid {
    color: red;
  }
}
