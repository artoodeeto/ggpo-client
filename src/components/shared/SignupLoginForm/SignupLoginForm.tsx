import React, { FC, FormEvent, useState } from 'react';
import { ILoginSignUpFormParams } from 'interfaces/session';
import { connect } from 'react-redux';
import { AnyAction } from 'redux';
import { ThunkDispatch } from 'redux-thunk';
import { RootState } from 'store/root/root_reducer';
import { logMeIn, signMeUp } from 'store/session/Actions';
import styles from './SignupLoginForm.module.scss';

const handleSubmit = (
  event: FormEvent,
  params: ILoginSignUpFormParams,
  signupOrLoginAPI: (userSignup: ILoginSignUpFormParams) => void
): void => {
  event.preventDefault();

  signupOrLoginAPI(params);
};

type SignupLoginFormProps = {
  toLoginOrSignup: boolean;
  onLogin: (userLogin: ILoginSignUpFormParams) => void;
  onSignup: (userSignup: ILoginSignUpFormParams) => void;
};

// TODO: change this as array of objects so we wont have render each input. we'll replace this with filter.
/**
 * TODO: sample object
 * [{
 *  onchange: fn,
 *  value,
 *  placeHolder,
 * type
 * }]
 */
export const SignupLoginForm: FC<SignupLoginFormProps> = ({ toLoginOrSignup, onLogin, onSignup }) => {
  const [email, setEmail] = useState('');
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');
  const [confirmPassword, setConfirmPassword] = useState('');
  const params = {
    email,
    username,
    password
  };

  return (
    <form
      className={styles.SignUpLoginForm}
      onSubmit={(e) => handleSubmit(e, params, toLoginOrSignup ? onLogin : onSignup)}
    >
      <input
        onChange={(e) => setEmail(e.target.value)}
        value={email}
        placeholder="email_address@gmail.com"
        type="text"
        name="email"
        id=""
        required
      />

      {toLoginOrSignup ? null : (
        <input
          onChange={(e) => setUsername(e.target.value)}
          value={username}
          placeholder="user name"
          type="text"
          name="username"
          id=""
          required
        />
      )}

      <input
        onChange={(e) => setPassword(e.target.value)}
        value={password}
        placeholder="password"
        type="text"
        name="password"
        id=""
        required
      />

      {toLoginOrSignup ? null : (
        <input
          onChange={(e) => setConfirmPassword(e.target.value)}
          value={confirmPassword}
          placeholder="confirm password"
          type="text"
          name="conf_password"
          id=""
          required
        />
      )}

      <input type="submit" value={toLoginOrSignup ? 'Login' : 'Signup'} />
    </form>
  );
};

const mapStateToProps = (state: RootState) => {
  return {};
};

const mapDispatchToProps = (dispatch: ThunkDispatch<RootState, {}, AnyAction>) => {
  return {
    onSignup: (userSignup: ILoginSignUpFormParams) => dispatch(signMeUp(userSignup)),
    onLogin: (userLogin: ILoginSignUpFormParams) => dispatch(logMeIn(userLogin))
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(SignupLoginForm);
