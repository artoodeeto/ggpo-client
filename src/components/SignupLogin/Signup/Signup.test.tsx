import React from 'react';
import SignupLoginForm from 'components/shared/SignupLoginForm/SignupLoginForm';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Signup } from './Signup';

describe('<Signup/>', () => {
  describe('Signup', () => {
    it('should show SignupLoginForm with props', () => {
      const wrapper = shallow(<Signup isLoggingInOrSigningUp={false} />);
      expect(shallowToJson(wrapper)).toMatchSnapshot();

      expect(wrapper.find(SignupLoginForm).exists()).toBeTrue();
      expect(wrapper.props()).toContainKeys(['toLoginOrSignup']);
    });
  });
});
