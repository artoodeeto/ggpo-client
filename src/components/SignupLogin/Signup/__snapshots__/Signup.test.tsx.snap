// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signup/> Signup should show SignupLoginForm with props 1`] = `
<Connect(SignupLoginForm)
  toLoginOrSignup={false}
/>
`;
